frontend/class-frontend.php

```diff
@@ -1147,6 +1147,8 @@ public function metakeywords() {
 			$keywords = WPSEO_Meta::get_value( 'metakeywords' );
 			if ( $keywords === '' && ( is_object( $post ) && ( ( isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== '' ) ) ) ) {
 				$keywords = wpseo_replace_vars( $this->options[ 'metakey-' . $post->post_type ], $post );
+			} else {
+				$keywords = wpseo_replace_vars( $keywords, $post );
 			}
```

Comment:
@andizer you're wrong, sorry.... :speak_no_evil:
Well, you're sort of right on the performance side or things. But you should only try to optimize performance if you can do so without compromising the functionality. You *are* however compromising the functionality (as you no longer have a fall-back if the keywords call would be empty after the replace call), which is why I commented on it.

> in my case a function call (to wpseo_replace_vars) will only be performed when $keywords and $post are not empty

Unfortunately not true as you don't have any checks in place for the metakeywords based call.
Still, nothing to worry about as the replace class is optimized for that and will return early if that's the case.

> In your case the function call will be executed twice when both values are empty

Actually no, if $post is empty, the second call will not be done anyway. See the `is_object( $post )` part of the `if()`.